generated/serialization.h

```diff
@@ -0,0 +1,80 @@
+#ifndef DTOGEN_SERIALIZATION_H
+#define DTOGEN_SERIALIZATION_H
+
+#include <memory>
+#include <optional>
+
+#include "nlohmann/json.hpp"
+
+// TODO: move elsewhere
+#ifdef __clang__
+#define UNREACHABLE_CODE __builtin_unreachable();
+#elif __GNUC__
+#define UNREACHABLE_CODE __builtin_unreachable();
+#elif _MSC_VER
+#define UNREACHABLE_CODE __assume(0);
+#endif
+
+// TODO: move elsewhere
+#include <stdexcept>
+namespace org {
+namespace openapitools {
+namespace model {
+class invalid_enum_value : public std::runtime_error {
+    public:
+        using std::runtime_error::runtime_error;
+};
+}  // namespace org
+}  // namespace openapitools
+}  // namespace model
+
+
+namespace nlohmann {
+
+template <typename T>
+struct adl_serializer<std::optional<T>> {
+    static void to_json(json& j, const std::optional<T>& opt) {
+        opt == std::nullopt ? j = nullptr : j = *opt;
+    }
+
+    static void from_json(const json& j, std::optional<T>& opt) {
+        j.is_null() ? opt = std::nullopt : opt = j.get<T>();
```

Comment:
Same as above. Probably that's just fine, but I would change it into a plain `if {} else` or use a "well formed" ternary:

```C++
opt = (!j.is_null()) ? j.get<T>() : std::nullopt
```

I didn't test the above, but I could imagine that it doesn't compile, because the two branches return different types. In that case you'd probably need to go more explicit:

```C++
opt = (!j.is_null()) ? std::optional<T>(j.get<T>()) : std::optional<T>()
```

Reply:
> I could imagine that it doesn't compile, because the two branches return different types

Exactly right. I will go with plain old `if`.